### PR TITLE
remove not implemented deprecated tests

### PR DESCRIPTION
**Pull request by christophe-papazian**
## Motivation

deprecated test should be removed

## Changes

Remove deprecated test
- Test_HttpClientIP
- Test_DistributedTraceInfo
- Test_TelemetryMetrics::test_metric_waf_updates

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by simon-id — APPROVED** [review]

**Review by nizox — APPROVED** [review]

File: tests/appsec/test_traces.py [hunk 1]
```diff
@@ -328,6 +328,7 @@ def validate_request_headers(span):
 class Test_DistributedTraceInfo:
     """Distributed traces info (Services, URL, trace id)"""
 
+    @irrelevant(True, reason="test non implemented")
```
Comment (cbeauchesne) [inline tests/appsec/test_traces.py:331]: In that case, juste remove the test. `irrelevant` can make think the test is not relevant.

File: tests/appsec/waf/test_telemetry.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def test_metric_waf_init(self):
 
     setup_metric_waf_updates = _setup
 
-    @missing_feature(reason="Test not implemented")
+    @irrelevant(reason="Test not implemented")
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_telemetry.py:80]: `missing_feature` is the good decorator, with the key principle : test is part of the feature.

Though, I wonder why there is such a comment. I mean, I can read some code

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/appsec/waf/test_telemetry.py (hunk 2 above)
Comment (christophe-papazian) [reply tests/appsec/waf/test_telemetry.py:80]: I'm checking with Santi about it...

File: tests/appsec/test_traces.py (hunk 1 above)
Comment (christophe-papazian) [reply tests/appsec/test_traces.py]: test removed.

File: tests/appsec/waf/test_telemetry.py (hunk 2 above)
Comment (christophe-papazian) [reply tests/appsec/waf/test_telemetry.py]: test removed.

**Review by smola — APPROVED** [review]
+1 for `test_metric_waf_updates`. I think I initially implemented this, then realized it required an RC update, and dropped the ball at the time. Ok to remove it.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**